core/src/misc.rs

```diff
@@ -30,6 +31,9 @@ pub struct MiscDetails {
 	/// Is this email a role-based account?
 	pub is_role_account: bool,
 	pub gravatar_url: Option<String>,
+	/// Is this email address listed in the haveibeenpwned database for
+	/// previous breaches?
+	pub haveibeenpwned: Option<CheckHaveIBeenPwnedDetails>,
```

Comment:
```suggestion
	pub haveibeenpwned: Option<u64>,
```

Let's also delete `CheckHaveIBeenPwnedDetails`, I don't think it's needed. If there are errors, let's just log them, no need to return the error to the user. See how it's done in gravatar.